pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -233,21 +145,71 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 	return registries, nil
 }
 
+func validateRegistries(regs []Registry) ([]Registry, error) {
+	var registries []Registry
+	regMap := make(map[string][]Registry)
+
+	for _, reg := range regs {
+		var err error
+
+		// make sure URL and Prefix are valid
+		reg.URL, err = parseURL(reg.URL)
+		if err != nil {
+			return nil, err
+		}
+
+		if reg.Prefix == "" {
+			reg.Prefix = reg.URL
```

Comment:
`s/validateRegistries/cleanAndValidateRegistries/`, or something like that, maybe?

It’s surprising for a “validate” function to modify its input.